aerotech_101smc2/aerotech_101smc2/_aerotech.py

```diff
@@ -36,21 +39,42 @@ def get_state(self):
         state["value"] = self.value
         return state
 
-
-
-
     def _set_position(self, position):
-        ...
-
-
+        self._serial_port.write(f"M {position}\n".encode())
+
+    def close(self):
+        self._serial_port.close()
+
+    def direct_serial_write(self, message):
+        self._busy = True
+        self._serial_port.write(message.encode())
+
+    def home(self):
+        # Since homing is typically a long process, start a new asynchronous task
+        # This may not be necessary, depending on how your device behaves,
+        # but remember that home is defined as returning to the current destination
+        # This method should return quickly, not wait for the homing to complete.
+        loop = asyncio.get_event_loop()
+        loop.create_task(self._home())
+
+    async def _home(self):
+        self._busy = True
+        # Initiate the home
+        self._serial_port.write(b"H\n")
+        await self._not_busy_sig.wait()
+        self.set_position(self._destination)
+    
     async def update_state(self):
-        """Continually monitor and update the current daemon state."""
-        # If there is no state to monitor continuously, delete this function
         while True:
             # Perform any updates to internal state
-            self._busy = False
-            # There must be at least one `await` in this loop
-            # This one waits for something to trigger the "busy" state
-            # (Setting `self._busy = True)
-            # Otherwise, you can simply `await asyncio.sleep(0.01)`
-            await self._busy_sig.wait()
+            self._serial_port.write(b"Q\n")
+            line = await self._serial_port.areadline()
+            self._busy = (line[0] != b"R")
+            self.logger.debug(line[0])
+            if self._busy:
+                await asyncio.sleep(0.1)
+                self._serial_port.write(b"G\n")
+                self._position = float(await self._serial_port.areadline())
```

Comment:
```suggestion
            self._busy = (line[0] != b"R")
            self.logger.debug(line[0])
            self._serial_port.write(b"G\n")
            self._position = float(await self._serial_port.areadline())
            if self._busy:
                await asyncio.sleep(0.1)

```

Reply:
In the case of not busy, this is exactly the same as what you suggest.
This, however, gives continuous feedback to users mid motion. Something that is particularly important for long stages.